test/functional/change_stream.test.js

```diff
@@ -2609,34 +2638,24 @@ describe('Change Streams', function() {
   describe('tryNext', function() {
     it('should return null on single iteration of empty cursor', {
       metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
-      test: function(done) {
-        const client = this.configuration.newClient();
-        client.connect(err => {
-          expect(err).to.not.exist;
-
-          const changeStream = client
-            .db()
-            .collection('test')
-            .watch();
-
+      test: function() {
+        return withTempDb('testTryNext', this.configuration.newClient(), db => done => {
```

Comment:
thoughts about the helpers:
- can `withClient` be modified to not need the `this.configuration.newClient` parameter if `this.configuration` exists?
```
withClient((client, done) => {
...
});
```
- I'm a little wary of functions that return functions (ad infinitum) in cases like this because it makes the test a bit hard to reason about. To be clear, I think what happens behind the scenes is totally fair game, but what we expose to the user (us) should be super intuitive, something I jump to use. What about composability:
```
withClient(withTempDb('testTryNext', (client, db, done) => {
  ... test ...
}));
```

Reply:
I agree it's worth thinking these helpers through a bit more. I had to make some modifications to `withMonitoredClient` in #2354 which could also be more elegant/usable.